compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -1112,7 +1112,10 @@ fn get_backend_from_raw_matches(
     matches: &Matches,
 ) -> Box<dyn CodegenBackend> {
     let debug_flags = matches.opt_strs("Z");
-    let backend_name = debug_flags.iter().find_map(|x| x.strip_prefix("codegen-backend="));
+    let backend_name = debug_flags.iter().find_map(|x| {
+        x.strip_prefix("codegen")
+            .and_then(|x| x.strip_prefix("-backend=").or_else(|| x.strip_prefix("_backend=")))
```

Comment:
This reads a bit clearer to me:

```suggestion
        x.strip_prefix("codegen-backend=").or(x.strip_prefix("codegen_backend="))
```

This code path is not performance sensitive at all.